client/my-sites/plugins/main.jsx

```diff
@@ -189,9 +197,17 @@ export default React.createClass( {
 		);
 	},
 
-	toggleBulkManagement() {
-		const bulkManagement = ! this.state.bulkManagement,
-			sites = bulkManagement ? this.props.sites.getSelectedOrAllWithPlugins() : this.props.sites.getSelectedOrAll();
+	toggleBulkManagement( section ) {
+		let bulkManagement = this.state.bulkManagement;
+		const foundIndex = bulkManagement.indexOf( section );
+
+		if ( foundIndex === -1 ) {
+			bulkManagement.push( section );
```

Comment:
Danger danger! React highly encourages _not_ modifying the `state` directly outside of `setState()` but this is exactly what the code is doing here. `bulkManagement` actually gets set to the _reference_ of `this.state.bulkManagement` and when we `push()` to it, we mutate the underlying array. A much safer way to do it would be via an immutable route.

```js
if ( foundIndex === -1 ) {
	bulkManagement = [ ...bulkManagement, section ];
	// or
	bulkManagement = bulkManagement.concat( section );
} else {
	bulkManagement = [ 
		...bulkManagement.slice( 0, foundIndex ),
		...bulkManagement.slice( foundIndex + 1 )
	];
	// or
	bulkManagement = []
		.concat( bulkManagement ) // clone the array
		.splice( foundIndex, 1 ); // remove the section
}
```